scripts/signedbuildskadi.sh

```diff
@@ -1,274 +1,311 @@
+
+  # Install Docker
+  sudo apt-get update
+  sudo apt-get install docker-ce -y
+  sudo systemctl enable docker
+
+  # Install Docker-Compose
+  sudo curl -L "https://github.com/docker/compose/releases/download/1.23.1/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+  sudo chmod +x /usr/local/bin/docker-compose
+  sudo curl -L https://raw.githubusercontent.com/docker/compose/1.23.1/contrib/completion/bash/docker-compose -o /etc/bash_completion.d/docker-compose
+
+  # Add skadi to docker usergroup
+  sudo usermod -aG docker $SKADI_USER
+}
+
+cdqr_cylr_setup () {
+  # Clone skadi and it's submodules
+  sudo git clone --recurse-submodules --branch $install_branch https://github.com/orlikoski/Skadi.git /opt/Skadi
+  sudo chown -R $SKADI_USER:$SKADI_USER /opt/Skadi
+
+  # Copy cdqr script to /usr/local/bin
+  sudo cp /opt/Skadi/scripts/cdqr /usr/local/bin/cdqr
+  sudo chmod +x /usr/local/bin/cdqr
+
+  # Create CyLR directory
+  sudo mkdir /opt/CyLR/
+  sudo chmod 777 /opt/CyLR
+
+  # Installs and Configures CDQR and CyLR
+  sudo -E bash /opt/Skadi/scripts/update.sh
+}
+
+timesketch_configs () {
+  # Write TS and Postgres creds to .env file
+  cd /opt/Skadi/Docker/
+  echo TIMESKETCH_USER=$TIMESKETCH_USER > ./.env
+  echo TIMESKETCH_PASSWORD=$TIMESKETCH_PASSWORD >> ./.env
+  echo POSTGRES_USER=$POSTGRES_USER >> ./.env
+  echo POSTGRES_PASSWORD=$psql_pw >> ./.env
+  echo HEAP_SIZE=1g >> ./.env
+
+  # Configure /etc/hosts file so the host can use same names for each service as the TimeSketch Dockers
+  echo 127.0.0.1       elasticsearch |sudo tee -a /etc/hosts
+  echo 127.0.0.1       postgres |sudo tee -a /etc/hosts
+  echo 127.0.0.1       neo4j |sudo tee -a /etc/hosts
+  echo 127.0.0.1       redis |sudo tee -a /etc/hosts
+
+  # Write TimeSketch config file on host
+  sudo cp /opt/Skadi/Docker/timesketch/timesketch.conf /etc/
+  sudo sed -i "s@SECRET_KEY = '<KEY_GOES_HERE>'@SECRET_KEY = '$SECRET_KEY'@g" /etc/timesketch.conf
+  sudo sed -i "s@<USERNAME>\:<PASSWORD>@$POSTGRES_USER\:$psql_pw@g" /etc/timesketch.conf
+  sudo sed -i "s@NEO4J_USERNAME = 'neo4j'@NEO4J_USERNAME = '$neo4juser'@g" /etc/timesketch.conf
+  sudo sed -i "s@NEO4J_PASSWORD = '<NEO4J_PASSWORD>'@NEO4J_PASSWORD = ''@g" /etc/timesketch.conf
+  sudo sed -i "s/UPLOAD_ENABLED = False/UPLOAD_ENABLED = True/g" /etc/timesketch.conf
+  sudo sed -i "s/GRAPH_BACKEND_ENABLED = False/GRAPH_BACKEND_ENABLED = True/g" /etc/timesketch.conf
+  sudo sed -i "s#@localhost/timesketch#@postgres/timesketch#g" /etc/timesketch.conf
+  sudo sed -i "s/ELASTIC_HOST = '127.0.0.1'/ELASTIC_HOST = 'elasticsearch'/g" /etc/timesketch.conf
+  sudo sed -i "s@'redis://127.0.0.1:6379'@'redis://redis:6379'@g" /etc/timesketch.conf
+  sudo sed -i "s/NEO4J_HOST = '127.0.0.1'/NEO4J_HOST = 'neo4j'/g" /etc/timesketch.conf
+
+  # Setup Nginx Auth
+  echo $NGINX_PASSWORD | sudo htpasswd -i -c /etc/nginx/.skadi_auth $NGINX_USER
+}
+
+clean_APT () {

```

Comment:
this should be called by the host setup (or just part of the host setup function itself). It's not needed anywhere else

Reply:
Updated this to be part of host_setup